### PR TITLE
[IMP] crm: harmonize won triggers with rainbowman

### PR DESCRIPTION
**Pull request by jeh-odoo**
# Purpose

Harmonize the rainbowman triggers when a lead is won.

# Specification

Now, the rainbowman and the message is displayed to the user
when he uses the statusbar to markk a lead as won or drag and 
drop a lead in the won stage.

task-2287758

### CODE REVIEW

File: addons/crm/static/src/js/crm_form.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    var FormController = require('web.FormController');
+    var FormView = require('web.FormView');
+    var viewRegistry = require('web.view_registry');
+
+    var CrmFormController = FormController.extend({
+        _applyChanges: async function (dataPointID, changes, event) {
+            var result = await this._super(...arguments);
+            if ('stage_id' in changes) {
+                var self = this;
+                this._rpc({
```
Comment (std-odoo) [inline addons/crm/static/src/js/crm_form.js:13]: Can you use `await` instead ?

```javascript
let message = await this._rpc({...});
```

File: addons/crm/static/src/js/crm_kanban.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    var KanbanModel = require('web.KanbanModel');
+    var KanbanView = require('web.KanbanView');
+    var viewRegistry = require('web.view_registry');
+
+    var CrmKanbanModel = KanbanModel.extend({
+        moveRecord: async function (recordID, groupID, parentID) {
+            var self = this;
+            var result = await this._super(...arguments);
+            this._rpc({
```
Comment (std-odoo) [inline addons/crm/static/src/js/crm_kanban.js:12]: You can also use await here, and so remove 
```javascript
var self = this;
```

File: addons/crm/static/src/js/crm_form.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            var result = await this._super(...arguments);
+            if ('stage_id' in changes) {
+                var self = this;
+                const message = await this._rpc({
+                    model: 'crm.lead',
+                    method: 'get_rainbowman_message',
+                    args: [event.target.res_id],
+                });
+                if (message) {
+                    self.trigger_up('show_effect',
```
Comment (std-odoo) [inline addons/crm/static/src/js/crm_form.js:19]: Can you use `this` instead ?

File: addons/crm/static/src/js/crm_form.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    var FormController = require('web.FormController');
+    var FormView = require('web.FormView');
+    var viewRegistry = require('web.view_registry');
+
+    var CrmFormController = FormController.extend({
+        _applyChanges: async function (dataPointID, changes, event) {
+            var result = await this._super(...arguments);
+            if ('stage_id' in changes) {
+                var self = this;
```
Comment (std-odoo) [inline addons/crm/static/src/js/crm_form.js:12]: I think you do not need `self` anymore

File: addons/crm/static/src/js/crm_kanban.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    "use strict";
+
+    var KanbanModel = require('web.KanbanModel');
+    var KanbanView = require('web.KanbanView');
+    var viewRegistry = require('web.view_registry');
+
+    var CrmKanbanModel = KanbanModel.extend({
+        moveRecord: async function (recordID, groupID, parentID) {
+            var result = await this._super(...arguments);
+            if (this.localData[parentID].groupedBy[0] === this.defaultGroupedBy[0]) {
```
Comment (std-odoo) [inline addons/crm/static/src/js/crm_kanban.js:11]: Maybe can you add a comment to explain that we are checking if the kanban view is grouped by "stage_id" ?

File: addons/crm/models/crm_lead.py [hunk 6]
```diff
@@ -701,7 +701,21 @@ def action_set_won_rainbowman(self):
...
+                'effect': {
+                    'fadeout': 'slow',
+                    'message': message,
+                    'img_url': '/web/image/%s/%s/image_1024' % (self.team_id.user_id._name, self.team_id.user_id.id) if self.team_id.user_id.image_1024 else '/web/static/src/img/smile.svg',
+                    'type': 'rainbow_man',
+                }
+            }
+        return True
+    
+    def get_rainbowman_message(self):
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py:716]: I think we could take the opportunity to make this method private

File: addons/crm/static/src/js/crm_form.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+odoo.define("crm.crm_form", function (require) {
```
Comment (tde-banana-odoo) [inline addons/crm/static/src/js/crm_form.js:1]: It's missing some docstring to explain a bit what you are doing :)

File: addons/crm/static/src/js/crm_kanban.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+odoo.define('crm.crm_kanban', function (require) {
```
Comment (tde-banana-odoo) [inline addons/crm/static/src/js/crm_kanban.js:1]: It's missing some docstring to explain a bit what you are doing :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase)ff

**Comment by tde-banana-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**